
On OmniOS, the nspr include directory (/usr/include/mps) is also the
nss include directory. This appears to be different from other systems
which place nspr headers under /usr/include/nspr.

The build system prioritises the nspr include directory over its own public
headers and so builds of nss pull in a mixture of headers from the
extracted source and the previous version installed on the live system.

This patch fixes that by re-ordering the headers so that bundled ones
always come first.

https://bugzilla.mozilla.org/show_bug.cgi?id=1819024

diff -wpruN '--exclude=*.orig' a~/coreconf/config.gypi a/coreconf/config.gypi
--- a~/coreconf/config.gypi	1970-01-01 00:00:00
+++ a/coreconf/config.gypi	1970-01-01 00:00:00
@@ -155,7 +155,6 @@
     },
     'standalone_static_library': 0,
     'include_dirs': [
-      '<(nspr_include_dir)',
       '<(nss_dist_dir)/private/<(module)',
     ],
     'conditions': [
diff -wpruN '--exclude=*.orig' a~/exports.gyp a/exports.gyp
--- a~/exports.gyp	1970-01-01 00:00:00
+++ a/exports.gyp	1970-01-01 00:00:00
@@ -14,6 +14,7 @@
           'direct_dependent_settings': {
             'include_dirs': [
               '<(nss_public_dist_dir)/nss',
+              '<(nspr_include_dir)',
             ]
           },
           'dependencies': [
diff -wpruN '--exclude=*.orig' a~/lib/dbm/src/src.gyp a/lib/dbm/src/src.gyp
--- a~/lib/dbm/src/src.gyp	1970-01-01 00:00:00
+++ a/lib/dbm/src/src.gyp	1970-01-01 00:00:00
@@ -20,6 +20,9 @@
         'hash_buf.c',
         'mktemp.c'
       ],
+      'include_dirs': [
+        '<(nspr_include_dir)',
+      ],
       'dependencies': [
         '<(DEPTH)/exports.gyp:dbm_exports'
       ]
diff -wpruN '--exclude=*.orig' a~/lib/util/util.gyp a/lib/util/util.gyp
--- a~/lib/util/util.gyp	1970-01-01 00:00:00
+++ a/lib/util/util.gyp	1970-01-01 00:00:00
@@ -39,6 +39,9 @@
         'utilmod.c',
         'utilpars.c'
       ],
+      'include_dirs': [
+        '<(nspr_include_dir)',
+      ],
       'dependencies': [
         '<(DEPTH)/exports.gyp:nss_exports'
       ]
